Sources/MongoSwift/MongoCollection+FindAndModify.swift

```diff
@@ -309,15 +313,16 @@ private class FindAndModifyOptions {
             }
         }
 
-        if extra.keys.count > 0 && !mongoc_find_and_modify_opts_append(self._options, extra.data) {
+        let appendSuccess = extra.keys.count <= 0 || mongoc_find_and_modify_opts_append(self._options, extra.data)
```

Comment:
so, a few things for this line, most of which are artifacts and not related to your change. but I'll do my best to explain why we should fix them because it's a good lesson in Swift and Swift-C performance.

so in general, it's preferred to use `isEmpty` on [`Collection`](https://developer.apple.com/documentation/swift/collection) types in Swift (which includes `Array`) rather than comparing `.count` to 0. in short, the reason for this is that you can usually check in O(1) if *something* is there, but it takes O(n) to count *how many* things are there. 

ok, so that means we should use `extra.keys.isEmpty`. BUT, that's actually still O(n).... reason being that [`Document.keys`](https://github.com/mongodb/mongo-swift-driver/blob/e98bdf5806458caf911db56aa5e7aed67d41b5bd/Sources/MongoSwift/BSON/Document.swift#L32) is a computed property and it's created on the fly when we ask for it, by iterating over the document and converting all the C string keys to Swift strings. 

so, the most optimal thing here is to use the recently added  [`Document.isEmpty`](https://github.com/mongodb/mongo-swift-driver/blob/e98bdf5806458caf911db56aa5e7aed67d41b5bd/Sources/MongoSwift/BSON/Document%2BSequence.swift#L20) property. this works by just asking libbson directly if there is at least one element in the underlying `bson_t`.

hopefully that all makes sense.... if not let me know.

ok, finally... prefer just line-breaking to deal with long lines rather than introducing new variables that will only be used once.

so basically, I can't encompass this in a suggestion, but let's make this line something like
```swift
guard extra.isEmpty || mongoc_find_and_modify_opts_append(self._options, extra.data) else { // ... 
```

Reply:
What you said makes perfect sense. TIL!!

The addition of `appendSuccess` wasn't made for the purpose of shortening the line length though. I did that because generally I tend to assign longer boolean expressions to variables so it becomes more readable. i.e. `appendSuccess` is more immediately obvious in meaning than `extra.keys.count <= 0 || mongoc_find_and_modify_opts_append(self._options, extra.data)`. However, it seems like this is not good Swift, so I've reverted that.

I've also included the suggestion to switch over to `extra.isEmpty`.